framework/src/play/mvc/Router.java

```diff
@@ -67,6 +68,10 @@ public static List<Route> routes() {
    */
   private static void loadRoutesFromFile() {
     instance.setRoutes(new RoutesParser().parse(Play.routes));
+    // This automatically added route is needed to serve the static files extracted from classpath.
+    // The "public" postfix (!) on the end is to prevent unwanted leakage of classpath resource files!
+    // TODO: automatically expose routes (e.g. with "staticDir:" parsing or with application.conf settings from dependencies instead of hardcoded "public"
+    addRoute("GET", "/public/", "staticDir:" + Play.tmpDir.getName() + "/" + SERVER_HELPER_FIND_FILE_TMP_PATH_PREFIX + "public");
```

Comment:
This dies with NPE when `play.tmpDir=none` is set in `application.conf`.
Fix is ready, only tests need to be created!